docs/VMR-Feature-Band-Support-Work-Plan.md

```diff
@@ -0,0 +1,139 @@
+    participant VMR_2xx as VMR<br />release/10.0.2xx
+    participant VMR_3xx as VMR<br />release/10.0.3xx
+
+    runtime->>VMR_1xx: Flow of runtime change
+    Note over VMR_1xx: 📦 VMR build output packages are built
+    VMR_1xx->>VMR_2xx: Dependency flow <br />(runtime build output packages)
+    VMR_1xx->>VMR_3xx: Dependency flow <br />(runtime build output packages)
+```
+
+This could be done a few different ways, but there are some explicit goals that point in a specific direction:
+- **Repeatability** - The product produced from the 2xx++ branches should not vary in a meaningful way if the source has not changed.
+- **Speed** - The time/number of steps required to build 2xx+ should be minimized.
+- **Distro partner/MSFT build unification** - We should minimize differences between the distro partner and MSFT builds.
+
+To achieve 2xx+ builds, we will use the "classic" dependency flow methodology that .NET has used for many years but with a variation to account for VMR-to-VMR flow. When flow happens from the the VMR's 1xx to 2xx branch, all dependencies across all relevant repos will be updated in one PR. This ensures coherency. The dependencies being updated are applied to the individual repos' Version.Details.xml files. There will not be a root VMR Version.Details.xml that aggregates all of these dependencies.
+
+Let's use roslyn's dependency on command-line-api as an example to demonstrate the flow.
+
+1xx branch behavior:
+1. command-line-api builds first
+1. roslyn gets the live version of command-line-api via PackageVersions.props overrides
+
+2xx branch behavior:
+1. VMR build of 1xx flows to 2xx
+  * roslyn's Version.Details updated to get the 1xx version of command-line-api
+1. roslyn build occurs
+  * No PackageVersions.props overrides happen because command-line-api doesn't exist in the build graph
+  * That's ok because the static version defined in Version.Details is the one we want to consume
+
+**Considerations**
+
+- **Validation of 2xx+ repos** - A repo is validated on backflow based on what it built against. So a 1xx SDK repo branch receives the source changes in the 1xx VMR sdk sources, as well as the outputs of the VMR build, which includes the runtime that the SDK built against in that VMR build. This means that on backflow, the SDK PR is validating the SDK's sources against the updated SDK dependencies. For 2xx+ branches, only a subset of outputs is produced, so a backflow PR would not include all the dependencies that the 2xx SDK needs. To deal with this, we will attempt to use the CoherentParentDependency feature. CoherentParentDependency attributes will be placed on the SDK dependencies that are not produced in the 2xx build, tying them to a dependency produced in the 2xx build. When backflow happens and the dependencies are updated for a repo, the CPDs will resolve to the versions specific in the root VMR Version.Details.xml if they are not produced in the current build.
```

Comment:
@mmitche - Is this still relevant? Since we'll represent all the dependencies each repo's V.D.xml instead of a root-level file, that V.D.xml info already does backflow to the product repo. So does this just work?

Reply:
No longer relevant based on the feature above.